pkg/template/processor_test.go

```diff
@@ -0,0 +1,604 @@
+package template_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	authv1 "github.com/openshift/api/authorization/v1"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var templateMetadata = `
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  labels:
+    provider: codeready-toolchain
+    version: ${COMMIT}
+  name: basic-tier-template
+objects:`
+
+var namespaceObj = `
+- apiVersion: v1
+  kind: Namespace
+  metadata:
+    annotations:
+      openshift.io/description: ${PROJECT_NAME}-user
+      openshift.io/display-name: ${PROJECT_NAME}
+      openshift.io/requester: ${USER_NAME}
+    labels:
+      provider: codeready-toolchain
+      version: ${COMMIT}
+    name: ${PROJECT_NAME}`
+
+var roleBindingObj = `
+- apiVersion: authorization.openshift.io/v1
+  kind: RoleBinding
+  metadata:
+    name: ${PROJECT_NAME}-edit
+    namespace: ${PROJECT_NAME}
+  roleRef:
+    kind: ClusterRole
+    name: edit
+  subjects:
+  - kind: User
+    name: ${USER_NAME}`
+
+var templateParams = `
+parameters:
+- name: PROJECT_NAME
+  required: true
+- name: USER_NAME
+  value: toolchain-dev
+  required: true
+- name: COMMIT
+  value: 123abc
+  required: true
+`
+
+var newUser = `
+  - kind: User
+    name: newUser
+`
+
+func TestProcess(t *testing.T) {
+	t.Run("should process template successfully", func(t *testing.T) {
+		// given
+		s := addToScheme(t)
+		namespace, commit, user := templateVars()
+		values := paramsKeyValues(namespace, commit, user)
+
+		cl := test.NewFakeClient(t)
+		p := template.NewProcessor(cl, s)
```

Comment:
I believe that the client, processor, and scheme can be declared out of the subtests as part of the test setup of the parent test (lines after 80) and then reuse it in all subtests

Reply:
done in 16e7265 for the `TestProcessor` func only (when testing the apply, we need to reset the client after each test, so it's simpler to declare a new one each time)